feature-libs/product-configurator/common/components/configurator-cart-entry-bundle-info/configurator-cart-entry-bundle-info.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, Optional } from '@angular/core';
+import { CartItemContext } from '@spartacus/storefront';
+import { EMPTY, Observable } from 'rxjs';
+import { OrderEntry } from '@spartacus/core';
+import { FormControl } from '@angular/forms';
+import { CommonConfiguratorUtilsService } from '../../shared/utils/common-configurator-utils.service';
+import { ConfiguratorCartEntryBundleInfoService } from './configurator-cart-entry-bundle-info.service';
+import { LineItem } from './configurator-cart-entry-bundle-info.model';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-configurator-cart-entry-bundle-info',
+  templateUrl: './configurator-cart-entry-bundle-info.component.html',
+})
+export class ConfiguratorCartEntryBundleInfoComponent {
+  constructor(
+    protected commonConfigUtilsService: CommonConfiguratorUtilsService,
+    protected configCartEntryBundleInfoService: ConfiguratorCartEntryBundleInfoService,
+    // TODO(#10946): make CartItemContext a required dependency and drop fallbacks to `?? EMPTY`.
+    @Optional() protected cartItemContext?: CartItemContext
+  ) {}
+
+  readonly orderEntry$: Observable<OrderEntry> =
+    this.cartItemContext?.item$ ?? EMPTY;
+
+  readonly quantityControl$: Observable<FormControl> =
+    this.cartItemContext?.quantityControl$ ?? EMPTY;
+
+  readonly readonly$: Observable<boolean> =
+    this.cartItemContext?.readonly$ ?? EMPTY;
+
+  numberOfLineItems$: Observable<number> = this.orderEntry$.pipe(
+    map((entry) =>
+      this.configCartEntryBundleInfoService.retrieveNumberOfLineItems(entry)
+    )
+  );
+
+  hideItems = true;
+
+  /**
+   * Toggles the state of the items list.
+   */
+  toggleItems(): void {
+    this.hideItems = !this.hideItems;
+  }
+
+  /**
+   * Retrieves the line items for an order entry.
+   *
+   * @param {OrderEntry} entry order entry
+   * @returns {LineItem[]} - array of line items
+   */
+  retrieveLineItems(entry: OrderEntry): LineItem[] {
+    return this.configCartEntryBundleInfoService.retrieveLineItems(entry);
+  }
+
+  /**
+   * Verifies whether the configurator type is bundle based one.
```

Comment:
Add an article maybe? Or remove the word 'one'.

Reply:
Done!